[CMake] Wrong linker being used for Mach-O cross compiling

I'm trying to build Darling on my Gentoo Linux. Clang is my default system compiler with custom patches. LLVM/LLD is my default ELF linker.

During building where it reaches the step it cross-compiles for Mach-O I noticed the wrong linker is being used resulting in funny error messages like this:

```plain
ld.lld: error: unknown argument '-dynamic', did you mean '-Bdynamic'
ld.lld: error: unknown argument '-dylib'
ld.lld: error: unknown argument '-arch'
ld.lld: error: unknown argument '-no_arch_warnings'
ld.lld: error: unknown argument '-dylib_install_name'
ld.lld: error: unknown argument '-compatibility_version'
ld.lld: error: unknown argument '-Z'
ld.lld: error: unknown argument '-dylib_file'
ld.lld: error: unknown argument '-dylib_file'
ld.lld: error: unknown argument '-dylib_file'
ld.lld: error: unknown argument '-dylib_file'
ld.lld: error: unknown argument '-dylib_file'
ld.lld: error: unknown argument '-dylib_file'
ld.lld: error: unknown argument '-dylib_file'
ld.lld: error: unknown argument '-dylib_file'
ld.lld: error: unknown argument '-dylib_file'
ld.lld: error: unknown argument '-dylib_file'
ld.lld: error: unknown argument '-dylib_file'
ld.lld: error: unknown argument '-dylib_file'
ld.lld: error: unknown argument '-dylib_file'
ld.lld: error: too many errors emitted, stopping now (use -error-limit=0 to see all errors)
```

I first was confused, but I figured out that the wrong linker is being invoked because I found out that the Apple ld was compiled earlier. I can workaround this problem my adding

```
set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fuse-ld=${CMAKE_BINARY_DIR}/src/external/cctools-port/cctools/ld64/src/x86_64-apple-darwin11-ld")
set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fuse-ld=${CMAKE_BINARY_DIR}/src/external/cctools-port/cctools/ld64/src/x86_64-apple-darwin11-ld")
```

to every CMake file which cross compiles for Mach-O, but that's pretty annoying.

Is this a bug or a incompatibility with my custom toolchain. I don't have GCC installed. My default "binutils" are the LLVM binutils.

```
$ clang -v
clang version 9.0.0 
Target: x86_64-pc-linux
Thread model: posix
InstalledDir: /usr/lib/llvm/9/bin
Found CUDA installation: /opt/cuda, version 10.1
```

My clang defaults to libc++ and compiler-rt.

I unfortunately hit a point where I can't continue :(

```
Linking C shared library libsystem_kernel.dylib
Undefined symbols for architecture x86_64:
  "___bzero", referenced from:
      _sigexc_handler in sigexc.o
     (maybe you meant: ____bzero)
ld: symbol(s) not found for architecture x86_64
```

I managed to get most of the libraries compiled and linked by now.

--- a/cmake/use_ld64.cmake
+++ b/cmake/use_ld64.cmake
@@ -1,5 +1,10 @@
 FUNCTION(use_ld64 target)
 	set_property(TARGET ${target} APPEND_STRING PROPERTY
+		COMPILE_FLAGS " -fuse-ld=${CMAKE_BINARY_DIR}/src/external/cctools-port/cctools/ld64/src/x86_64-apple-darwin11-ld ")
+	set_property(TARGET ${target} APPEND_STRING PROPERTY
+		LINK_FLAGS " -fuse-ld=${CMAKE_BINARY_DIR}/src/external/cctools-port/cctools/ld64/src/x86_64-apple-darwin11-ld ")
+
+	set_property(TARGET ${target} APPEND_STRING PROPERTY
 		LINK_FLAGS " -B ${CMAKE_BINARY_DIR}/src/external/cctools-port/cctools/ld64/src/ \
 -B ${CMAKE_BINARY_DIR}/src/external/cctools-port/cctools/misc/ \
 -target x86_64-apple-darwin11 -Wl,-Z \

